### PR TITLE
improvement(argus): Add linter and formatter for JavaScript

### PR DESCRIPTION
**Pull request by k0machi**
This commit adds ESLint for linting and enforcing a code style and
Prettier code formatter for formatting files. It also adds matching
configurations to them.

### CODE REVIEW

**Comment by fruch** [comment]
@k0machi how about adding a github action for running it ? (or even adding pre-commit, and running it on an action) ?

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module.exports = {
+    "env": {
+        "browser": true,
+        "node": true,
+        "es2021": true
+    },
+    "extends": [
+        "eslint:recommended",
```
Comment (fruch) [inline .eslintrc.js:8]: isn't it better to pin those down ? to not get surprises ?

**Comment by fruch** [comment]
@k0machi all the code is passing this reformatter and lint ? or those changes would be in a new PR/commit ?

File: .eslintrc.js (hunk 1 above)
Comment (k0machi) [reply .eslintrc.js:8]: It's a lot of rules that have to be defined explicitly and they're pretty universal, nothing opinionated.

**Comment by k0machi** [comment]
> @k0machi all the code is passing this reformatter and lint ? or those changes would be in a new PR/commit ?

That would be next PR/commit, I still need to run it initially and fix all the problems in all files that crop up.

File: .eslintrc.js (hunk 1 above)
Comment (fruch) [reply .eslintrc.js:8]: I'm just saying it might be changed from time to time, it doesn't have versions ?

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**